### PR TITLE
In 957 configure django auth

### PR DESCRIPTION
**Pull request by jonavellecuerdo**
### What does this PR do?

Refactor the Solenoid application to use Django’s built-in authentication system to return functionality to the application.

#### Helpful background context

* To configure authentication with Django, I closely followed [this guide](https://developer.mozilla.org/en-US/docs/Learn/Server-side/Django/Authentication).
* Users will not be prompted to change their password when logging in for the first time (I didn't want to spend too much time trying to get this to work), _however_ (and more importantly), **users can change their password after signing in.**
* When running the Django application locally, a reset link was only successfully generated if the `User.email` was provided.
* Password validators were configured for the Django app using [this guide](https://docs.djangoproject.com/en/4.2/topics/auth/passwords/#enabling-password-validation). 
   * Please refer to the doc to get a better understanding of the validators used to confirm password strength.
   * This is IS&T's [guide for strong passwords](http://kb.mit.edu/confluence/display/istcontrib/Strong+Passwords). Some of the standard validators configured for the app coincide with the requirements for IS&T, but it will not be a 1-to-1 match. However, I think it is sufficient for now. 
* To create user accounts, the following steps were taken:
   * Download Heroku CLI
   * Log into Heroku
   * Create a superuser account (i.e., admin) with the command:
      ```
      heroku run -a=mitlibraries-in-957-con-ytpnes python manage.py createsuperuser
      ```
   * To create users, [the admin panel](https://developer.mozilla.org/en-US/docs/Learn/Server-side/Django/Authentication#creating_users_and_groups) was used.

### How can a reviewer see these changes?

Opening this PR created a review app on Heroku. 

1. Visit https://mitlibraries-in-957-con-ytpnes.herokuapp.com/. 
2. Confirm that you can log in successfully using the credentials shared with you (via LastPass).
3. Confirm that you can log out successfully.
4. Confirm that you can reset a forgotten password.
   * From the login page, select `Need help signing in?`
   * You should see/receive a link to reset your password.
   * Fill out the form with a new password.
   * Confirm that you can log in successfully using the new password. 
6. Confirm that you can change your password at anytime after you are logged in. 
   * After logging in, select `Change my password` from the top of the homepage.
   * Fill out the form with your old and new passwords.
      * **Note:** Check what happens when you provide an input that doesn't meet the password requirements (e.g., type in '<your username>', '123', '123456789').
   * Confirm that you can log in successfully using the new password. 

### Includes new or updated dependencies?
YES

### Reviewer checklist
- [ ] The commit message is clear and follows our guidelines
- [ ] There are tests covering any new functionality
- [ ] The documentation has been updated if necessary
- [ ] The changes, if applicable, have been verified

### CODE REVIEW

File: solenoid/settings/base.py [hunk 1]
```diff
@@ -1,430 +1,293 @@
...
-
-# INTERNATIONALIZATION CONFIGURATION
-# -----------------------------------------------------------------------------
-
-# https://docs.djangoproject.com/en/1.8/topics/i18n/
-
-LANGUAGE_CODE = 'en-us'
-
-TIME_ZONE = 'UTC'
+# INTERNALIZATION
```
Comment (ehanson8) [inline solenoid/settings/base.py:144]: `# INTERNATIONALIZATION`?

**Review by ehanson8 — COMMENTED** [review]
This works as expected and I don't see any major problems but I'll defer to Graham for final approval

File: solenoid/accounts/templates/accounts/logout.html [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{% extends "base.html" %}
+
+{% block content %}
+  <p>You have successfully logged out. Have a nice day! </p>
```
Comment (ghukill) [inline solenoid/accounts/templates/accounts/logout.html:4]: 😄

File: solenoid/urls.py [hunk 3]
```diff
@@ -22,24 +22,29 @@
...
+        include("solenoid.accounts.urls", namespace="accounts"),
+    ),
+    re_path(r"^$", HomeView.as_view(), name="home"),
+    re_path(
+        r"^celery-progress/", include("celery_progress.urls", namespace="celery_progress")
+    ),
+    re_path(r"^records/", include("solenoid.records.urls", namespace="records")),
+    re_path(r"^emails/", include("solenoid.emails.urls", namespace="emails")),
+    re_path(r"^people/", include("solenoid.people.urls", namespace="people")),
+    # re_path(r'^oauth2/', include('social_django.urls', namespace='social')),
```
Comment (ghukill) [inline solenoid/urls.py:37]: Normally I would suggest removing this commented out code, but perhaps there is some utility in keeping as a breadcrumb while the app is being revitalized?  Not a request here, just noting.

File: solenoid/urls.py [hunk 4]
```diff
@@ -22,24 +22,29 @@
...
+    # re_path(r'^logout/$',
+    #         SolenoidLogoutView.as_view(template_name='userauth/logout.html'),
+    #         name='logout'),
```
Comment (ghukill) [inline solenoid/urls.py:40]: Do you think there is value in having `/logout` (confirmed does not work) redirect to view `/accounts/logout` (confirmed works)?   Not a request, just an option.

**Review by ghukill — APPROVED** [review]
Login, logout, password change, all these things are working as expected.

Nice work!  This is definitely a non-trivial task to take a non-functional app with external auth requirements and rework for Django auth.  

Left a couple of comments, but they are _purely_ questions or suggestions, no requested changes; feel free to merge given this approval.

One last thing: I noticed it looks like the app is running in DEBUG mode?  

Screenshot 2023-11-29 at 4 10 12 PM

Running a Django app in DEBUG mode is not ideal in production (obviously), but perhaps there are CI settings that will toggle this off for production?   or, perhaps we are okay with this for some reason?

A couple of places this could be set:
- https://github.com/MITLibraries/solenoid/blob/IN-957-configure-django-auth/app.json#L8
- https://github.com/MITLibraries/solenoid/blob/IN-957-configure-django-auth/app.json#L8
  - where this suggests it WILL get toggled off in prod

Overall, looks great!  Approved!

File: solenoid/settings/base.py (hunk 1 above)
Comment (jonavellecuerdo) [reply solenoid/settings/base.py:144]: Hmm, [this](https://github.com/MITLibraries/solenoid/blob/main/solenoid/settings/base.py#L126C2-L126C2) was the original comment header 🤔. I think the comments were to indicate the category of each setting: https://docs.djangoproject.com/en/4.2/topics/i18n/!

File: solenoid/urls.py (hunk 3 above)
Comment (jonavellecuerdo) [reply solenoid/urls.py:37]: I removed it to avoid confusion in the future!

File: solenoid/urls.py (hunk 4 above)
Comment (jonavellecuerdo) [reply solenoid/urls.py]: Thank you for the suggestion! The change has been made.

File: solenoid/settings/base.py (hunk 1 above)
Comment (jonavellecuerdo) [reply solenoid/settings/base.py]: Oooh, 😅 . I fixed it!

**Comment by ehanson8** [comment]
> * Side note: When you follow step 4 above, do ya'll receive 2 emails? 😅

No, just 1

**Comment by jonavellecuerdo** [comment]
Ah! I see. I think I was getting two because I used my email for the 'admin' user account as well. 😅 All is good now!

**Review by ghukill — APPROVED** [review]

**Pull request merged**